server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -198,7 +198,24 @@ func mirrorToCache(ctx context.Context, bsClient bspb.ByteStreamClient, remoteIn
 	if rsp.StatusCode < 200 || rsp.StatusCode >= 400 {
 		return nil, status.UnavailableErrorf("failed to fetch %q: HTTP %s", uri, err)
 	}
-	// Download to a temp file to avoid running out of memory.
+
+	// If we know what the SHA256 should be and the content length is known,
+	// then we know the full digest, and can pipe directly from the HTTP
+	// response to cache.
+	if expectedSHA256 != "" && rsp.ContentLength >= 0 {
```

Comment:
We are no longer validating the digest before we write the data to the cache. Should that be a requirement?

Reply:
It's not needed in this case since the ByteStreamServer will validate the digest before committing to the cache.

Reply:
Ack